File: src/test/java/dev/retrotv/crypt/owe/sha/SHA3512Test.java
package dev.retrotv.crypt.owe.sha;

import dev.retrotv.crypt.owe.OWETest;
import dev.retrotv.crypt.sha.SHA3512;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

public class SHA3512Test extends OWETest {

    @Test
    @DisplayName("SHA3512 password encode 테스트")
    void passwordEncrypt() {
        passwordEncryptAndMatchesTest(new SHA3512());
    }
}

File: src/test/kotlin/dev/retrotv/crypto/owe/hash/sha/SHA3512Test.kt
package dev.retrotv.crypto.owe.hash.sha

import dev.retrotv.crypto.owe.hash.OWETest
import dev.retrotv.enums.Algorithm.Hash.SHA3512
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

internal class SHA3512Test : OWETest() {
    @Test
    @DisplayName("SHA3512 File hash 테스트")
    @Throws(Exception::class)
    fun fileHashTest() {
        fileHashTest(SHA3512)
    }

    @Test
    @DisplayName("SHA3512 File hash matches 테스트")
    @Throws(Exception::class)
    fun fileHashMatchesTest() {
        fileHashMatchesTest(SHA3512(), SHA3512)
    }

    @Test
    @DisplayName("SHA3512 File and File matches 테스트")
    @Throws(
        Exception::class
    )
    fun fileMatchesTest() {
        fileMatchesTest(SHA3512())
    }

    @Test
    @DisplayName("SHA3512 password encode 테스트")
    fun passwordEncrypt() {
        passwordEncryptAndMatchesTest(SHA3512())
    }
}
